app/controllers/concerns/activate_navigation.rb

```diff
@@ -7,43 +7,79 @@ module ActivateNavigation
-      "/events/#{current_event.slug}/staff/program/proposals" => "event-program-proposals-link",
-      "/events/#{current_event.slug}/staff/program/sessions" => "event-program-sessions-link",
-      "/events/#{current_event.slug}/staff/program/speakers" => "event-program-speakers-link",
+    return 'active' if matches_nav_path?(key, program_subnav_item_map)
+  end
+
+  def schedule_subnav_item_class(key)
+    {} #TBD
+  end
+
+  def matches_nav_path?(key, paths={})
+    if paths.is_a?(String)
+      return paths==request.path
+
+    elsif paths.is_a?(Hash)
+      return matches_nav_path?(nil, paths.values) if key.nil?
+      return paths.any?{|k, p| k==key && matches_nav_path?(nil, p) } # only recurse if the key matches
+
+    elsif paths.is_a?(Array) || key.nil?
+      return paths.any?{|p| !p.nil? && matches_nav_path?(nil, p) }
+    end
+  end
+
+  def nav_item_map
+    @nav_item_map ||= {
+        'my-proposals-link' => add_path(:proposals),
+        'event-review-proposals-link' => [
+            add_path(:event_staff_proposals, current_event),
+            add_path(:event_staff_proposal, current_event, @proposal)
+        ],
+        'event-program-link' => program_subnav_item_map.values,
+        'event-schedule-link' => add_path(:event_staff_time_slots, current_event),
+        'event-dashboard-link' => event_subnav_item_map.values,
     }
-    return "active" if key == map[request.path]
+  end
+
+  def event_subnav_item_map
+    @event_subnav_item_map ||= {
+        'event-staff-dashboard-link' => add_path(:event_staff, current_event),
+        'event-staff-info-link' => add_path(:event_staff_info, current_event),
+        'event-staff-teammates-link' => add_path(:event_staff_teammates, current_event),
+        'event-staff-config-link' => add_path(:event_staff_config, current_event),
+        'event-staff-guidelines-link' => add_path(:event_staff_guidelines, current_event),
+        'event-staff-speaker-emails-link' => add_path(:event_staff_speaker_email_notifications, current_event),
+    }
+  end
+
+  def program_subnav_item_map
+    @program_subnav_item_map ||= {
+        'event-program-proposals-selection-link' => [
+            add_path(:selection_event_staff_program_proposals, current_event),
+            # add_path(:event_staff_program_proposal, current_event, @proposal)
+            #How to leverage session[:prev_page] here ?
+        ],
+        'event-program-proposals-link' => [
+            add_path(:event_staff_program_proposals, current_event),
+            add_path(:event_staff_program_proposal, current_event, @proposal)
+        ],
+        'event-program-sessions-link' => add_path(:event_staff_program_sessions, current_event),
+        'event-program-speakers-link' => add_path(:event_staff_program_speakers, current_event),
+    }
+  end
+
+  def add_path(sym, *deps)
+    send(sym.to_s + '_path', *deps) unless deps.include?(nil) # don't generate the path unless all dependencies are present
```

Comment:
Couple things I don't like about this:
1) this is a controller concern, so this should definitely be declared private.
2) converting the sym to a string uses a bit of memory each time. Why not just pass in the whole name of the helper as a string? Readability with the symbols is nice though, and if the method can't be made private, it's definitely safer than having a random string + dependencies passed in. Confirm that a concern's private methods keep their privateness when they're included